### ISSUE TITLE
sort-prop-types: follow variable reference

### ISSUE THREAD

**Issue by oliviertassinari** [yannickcr/eslint-plugin-react#622]
As discussed in this [failing test case](https://github.com/yannickcr/eslint-plugin-react/pull/567)
I think that it would be great to support the following type of pattern:

``` jsx
const propTypes = {
  muiTheme: PropTypes.object.isRequired,
  className: PropTypes.string,
};

const TextFieldLabel = (props) => {
  return (
    <div />
  );
};

TextFieldLabel.propTypes = propTypes
```

However, it's not simple, it requires to follow variable reference.

**Comment by ljharb** [comment]
When it's in the same file, this absolutely should work imo.

**Comment by Bouncey** [comment]
:+1: Would love this

**Comment by Bouncey** [comment]
Great news @yannickcr!! :medal_sports: 

The next question is obviously about the next release :+1: :wink:

**Comment by yannickcr** [comment]
I hope to publish a new release tomorrow :)
